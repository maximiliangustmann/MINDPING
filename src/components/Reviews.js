import React, { useState } from 'react'
import Filter from './Filter'
import Review from './Review'

export default function Reviews({ reviews, onRemove, onEdit }) {
  const filters = ['Show all', 'book', 'audiobook', 'podcast', 'film', 'series']
  const [active, setActive] = useState(filters[0])
  return (
    <>
      <Filter filters={filters} active={active} setActive={setActive} />
      {reviews
        ?.filter(
          (reviews) =>
            active === 'Show all' || reviews.category.includes(active)
        )
        .map((review) => (
          <Review
            key={review.id}
            {...review}
            onRemove={onRemove}
            onEdit={onEdit}
          />
        ))}
    </>
  )
}
