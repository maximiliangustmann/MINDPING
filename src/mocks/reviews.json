[
  {
    "title": "Kangoroo Chronicles",
    "rating": "8.0",
    "categorie": "Book",
    "subcategorie": "Fiction",
    "summary": "Living together with a communistic kangoroo",
    "lessons": "Questioning existing structures we assume as irreversible"
  },
  {
    "title": "The 4-hour work week",
    "rating": "9.7",
    "categorie": "Book",
    "subcategorie": "Non-fiction",
    "summary": "Guide to life your dreamlife",
    "lessons": "Living more and working less"
  }
]
