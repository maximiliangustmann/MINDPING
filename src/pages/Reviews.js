import React from 'react'
import Filter from '../components/Filter'
import Review from '../components/Review'
import Header from '../components/Header'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

Reviews.propTypes = {
  reviews: PropTypes.array,
  onRemove: PropTypes.func,
  onEdit: PropTypes.func,
  filters: PropTypes.array,
  active: PropTypes.string,
  setActive: PropTypes.func,
  activeFilter: PropTypes.func,
  sortBySubcategory: PropTypes.func,
  sortByRating: PropTypes.func,
}

export default function Reviews({
  reviews,
  onRemove,
  onEdit,
  filters,
  active,
  setActive,
  activeFilter,
  sortBySubcategory,
  sortByRating,
}) {
  return (
    <>
      <Header
        sortBySubcategory={sortBySubcategory}
        sortByRating={sortByRating}
      />
      <StyledReviews>
        {reviews
          ?.filter(
            (reviews) =>
              active === 'Show all' || reviews.category.includes(active)
          )
          .map((review) => (
            <Review
              key={review.id}
              {...review}
              onRemove={onRemove}
              onEdit={onEdit}
            />
          ))}
      </StyledReviews>

      <Filter
        filters={filters}
        active={active}
        setActive={setActive}
        activeFilter={activeFilter}
      />
    </>
  )
}

const StyledReviews = styled.div`
  display: grid;
  gap: 20px;
  @media (min-width: 750px) and (max-width: 1124px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
  }
  @media (min-width: 1125px) {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
  }
`
