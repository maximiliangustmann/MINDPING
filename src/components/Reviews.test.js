import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Reviews from './Reviews'
import reviews from '../mocks/reviews.json'

describe('Reviews', () => {
  it('renders all properties', () => {
    render(<Reviews reviews={reviews} />)
  })
  it('renders correctly', () => {
    const tree = renderer.create(<Reviews />)
    expect(tree).toMatchSnapshot()
  })
})
