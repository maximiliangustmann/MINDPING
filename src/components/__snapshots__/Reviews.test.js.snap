// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews renders correctly 1`] = `
Array [
  <section
    className="Reviews__Review-ejni7u-0 ebDGLA"
  >
    <h2>
      Kangoroo Chronicles
    </h2>
    <p>
      Rating: 
      8.0
    </p>
    <h3>
      Book
    </h3>
    <h3>
      Fiction
    </h3>
    <h3>
      Summary: 
    </h3>
    <p>
      Living together with a communistic kangoroo
    </p>
    <h3>
      Lessons: 
    </h3>
    <p>
      Questioning existing structures we assume as irreversible
    </p>
  </section>,
  <section
    className="Reviews__Review-ejni7u-0 ebDGLA"
  >
    <h2>
      The 4-hour work week
    </h2>
    <p>
      Rating: 
      9.7
    </p>
    <h3>
      Book
    </h3>
    <h3>
      Non-fiction
    </h3>
    <h3>
      Summary: 
    </h3>
    <p>
      Guide to life your dreamlife
    </p>
    <h3>
      Lessons: 
    </h3>
    <p>
      Living more and working less
    </p>
  </section>,
]
`;
